### PR TITLE
Accept path while taking picture in android

### PR DESCRIPTION
**Pull request by AsminBudha**
# Summary
With this PR, user are able to add a path options while taking picture same as recording picture. It will save image in given path i.e if it is file path then picture will be save with give in that location else if it is directory then name will be randomly generated.
 It impact the photo taking process.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ❌     |
| Android |    ✅     |

## Checklist

- [X] I have tested this on a device and a simulator
- [X] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [ ] I updated the typed files (TS and Flow)
- [ ] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

File: docs/RNCamera.md [hunk 1]
```diff
@@ -598,6 +599,8 @@ writeExif = {
 
 - `orientation` (string or number). Specifies the orientation that us used for taking the picture. Possible values: `"portrait"`, `"portraitUpsideDown"`, `"landscapeLeft"` or `"landscapeRight"`.
 
+- `path` (file path on disk). Specifies the path on disk to save picture to.
```
Comment (MateusAndrade) [inline docs/RNCamera.md]: What about warning that setting this, can slow down the proccess of taking a picture?

Comment (AsminBudha) [reply docs/RNCamera.md]: Without path option, it saves in Android/data/{app_package_name}/files so I think it won't much affect the speed if picture is save in another location.

File: docs/RNCamera.md [hunk 2]
```diff
@@ -599,7 +599,7 @@ writeExif = {
 
 - `orientation` (string or number). Specifies the orientation that us used for taking the picture. Possible values: `"portrait"`, `"portraitUpsideDown"`, `"landscapeLeft"` or `"landscapeRight"`.
 
-- `path` (file path on disk). Specifies the path on disk to save picture to.
+- `path` (file path on disk). Specifies the path on disk to save picture to. Picture taking process might be slower as picture is saved other than cache directory.
```
Comment (sibelius) [inline docs/RNCamera.md]: Is it?

Do we have a benchmark for this?

Comment (AsminBudha) [reply docs/RNCamera.md]: I did check on by seeing difference in System.time between starting of taking photo process to completion of photo taking process. And, cache directory was about 100ms faster.

File: docs/RNCamera.md (hunk 1 above)
Comment (AsminBudha) [reply docs/RNCamera.md]: @MateusAndrade @sibelius 
Sorry for confusion but my above test was an fault might be due to setting debugger on. Now, I think it won't affect the speed as in both cases we have to write file into directory.

**Comment by cristianoccazinsp** [comment]
Hey guys, I think this PR introduced a bug.

The following code was updated: https://github.com/AsminBudha/react-native-camera/blob/09823726f38e755220a14fb2e376a3a272ebbe05/android/src/main/java/org/reactnative/camera/tasks/ResolveTakenPictureAsyncTask.java#L184 

However, the alternative case (in the else clause) wasn't updated: https://github.com/AsminBudha/react-native-camera/blob/09823726f38e755220a14fb2e376a3a272ebbe05/android/src/main/java/org/reactnative/camera/tasks/ResolveTakenPictureAsyncTask.java#L238

I believe the same update needs to be applied to https://github.com/AsminBudha/react-native-camera/blob/09823726f38e755220a14fb2e376a3a272ebbe05/android/src/main/java/org/reactnative/camera/tasks/ResolveTakenPictureAsyncTask.java#L324

**Comment by sibelius** [comment]
can @AsminBudha or @cristianoccazinsp send a PR to fix this?

**Comment by cristianoccazinsp** [comment]
I would like @AsminBudha  to take a look first, since although there's a confirmed bug/inconsistency there, it still doesn't explain this other bug report: https://github.com/react-native-community/react-native-camera/issues/2778

I will try to test this now that I'm preparing a PR to fix https://github.com/react-native-community/react-native-camera/issues/2780

**Comment by sibelius** [comment]
tks

**Comment by cristianoccazinsp** [comment]
I'm working on a PR to fix this bug and also implement this for iOS.

**Pull request merged**